mne/decoding/transformer.py

```diff
@@ -66,24 +100,38 @@ def fit(self, epochs_data, y):
                              % type(epochs_data))
 
         X = np.atleast_3d(epochs_data)
+        n_epochs, n_channels, n_times = X.shape
+        X = X.transpose(1, 0, 2)
+        self.ch_mean_ = np.zeros(n_channels)
+        self.scale_ = np.zeros(n_channels)
 
-        picks_list = dict()
-        picks_list['mag'] = pick_types(self.info, meg='mag', ref_meg=False,
-                                       exclude='bads')
-        picks_list['grad'] = pick_types(self.info, meg='grad', ref_meg=False,
-                                        exclude='bads')
-        picks_list['eeg'] = pick_types(self.info, eeg=True, ref_meg=False,
-                                       meg=False, exclude='bads')
+        if self.scalings == "mean":
+            from sklearn.preprocessing import StandardScaler
 
-        self.picks_list_ = picks_list
+            std_scaler = StandardScaler(self.with_mean, self.with_std)
+            self._pick_apply(X, std_scaler, n_channels)
 
-        for key, this_pick in picks_list.items():
-            if self.with_mean:
-                ch_mean = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.ch_mean_[key] = ch_mean  # TODO rename attribute
-            if self.with_std:
-                ch_std = X[:, this_pick, :].mean(axis=1)[:, None, :]
-                self.std_[key] = ch_std  # TODO rename attribute
+        elif self.scalings == 'median':
+            from sklearn.preprocessing import RobustScaler
+
+            rob_scaler = RobustScaler(self.with_centering, self.with_std)
+            self._pick_apply(X, rob_scaler, n_channels)
```

Comment:
I think this is still not the right syntax... You're using multiple lines to do the same thing. More importantly, the _pick_apply() function looks pretty weird to me, and the 'mean' and 'median' option need not picking, right?

How about you simplify all this with something like
```python
if scalings == 'mean': 
    scaler = StandardScaler(...) 
elif scalings == 'median':
    scaler = RobustScaler(...) 
elif scalings == 'default_constant':
    class ConstantScaler():
         def fit(X, y=None):
             self.std_ = np.array([scaling_dict[typ] for typ in ch_types])
             return self
         def transform(X):
             return X / self.std_
         def fit_transform(X, y=None):
             return self.transform(X)
     scaler = ConstantScaler()

self._scaler = scaler

...
self.transform(X):
    return self._scaler.transform(X)
```